File: GetVariables.cc
#include <iostream>
#include <string>
#include "TFile.h"
#include "TChain.h"
#include "TTree.h"
using namespace std;

void GetVariables(){
  bool passedTring;
  bool passedZ1LSelection;
  bool passedZ4lSelection;
  bool passedFullSelection;
  std::vector<int> *lep_id;
  std::vector<float> *lep_pt;
  std::vector<float> *lep_eta;
  std::vector<float> *lep_phi;
  std::vector<float> *lep_mass;
  std::vector<float> *lepFSR_pt;
  std::vector<float> *lepFSR_eta;
  std::vector<float> *lepFSR_phi;
  std::vector<float> *lepFSR_mass;
  Int_t finalState;
  Int_t lep_Hindex;
  vector<float>   *Z_pt;
  vector<float>   *Z_eta;
  vector<float>   *Z_phi;
  vector<float>   *Z_mass;
  vector<float>   *Z_noFSR_pt;
  vector<float>   *Z_noFSR_eta;
  vector<float>   *Z_noFSR_phi;
  vector<float>   *Z_noFSR_mass;
  std::vector<float> *H_pt;
  std::vector<float> *H_eta;
  std::vector<float> *H_phi;
  std::vector<float> *H_mass;

  TString outputFile = "Fourlep_2018_01.root";
  TFile ofile (outputFile,"recreate");
  TTree* newtree = new TTree("newtree","newtree");
  newtree->Branch("passedZ1LSelection", &passedZ1LSelection);
  newtree->Branch("passedFullSelection", &passedFullSelection);
  newtree->Branch("finalState",&finalState);
  newtree->Branch("passedZ4lSelection", &passedZ4lSelection);
  newtree->Branch("lep_id", &lep_id);
  newtree->Branch("lep_pt", &lep_pt);
  newtree->Branch("lep_eta", &lep_eta);
  newtree->Branch("lep_phi", &lep_phi);
  newtree->Branch("lep_mass", &lep_mass);
  newtree->Branch("lepFSR_pt", &lepFSR_pt);
  newtree->Branch("lepFSR_eta", &lepFSR_eta);
  newtree->Branch("lepFSR_phi", &lepFSR_phi);
  newtree->Branch("lepFSR_mass", &lepFSR_mass);
  newtree->Branch("Z_pt",&Z_pt);
  newtree->Branch("Z_eta",&Z_eta);
  newtree->Branch("Z_phi",&Z_phi);
  newtree->Branch("Z_mass",&Z_mass);
  newtree->Branch("Z_noFSR_pt",&Z_noFSR_pt);
  newtree->Branch("Z_noFSR_eta",&Z_noFSR_eta);
  newtree->Branch("Z_noFSR_phi",&Z_noFSR_phi);
  newtree->Branch("Z_noFSR_mass",&Z_noFSR_mass);
  newtree->Branch("H_pt",&H_pt);
  newtree->Branch("H_eta",&H_eta);
  newtree->Branch("H_phi",&H_phi);
  newtree->Branch("H_mass",&H_mass);

  TChain* chain = new TChain("Ana/passedEvents");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/SingleMuon/crab_SingleMuon_Run2018A-17Sep2018-v2/201018_142705/0000/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/SingleMuon/crab_SingleMuon_Run2018A-17Sep2018-v2/201018_142705/0001/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/SingleMuon/crab_SingleMuon_Run2018A-17Sep2018-v2/201018_142705/0002/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/DoubleMuon/crab_DoubleMuon_Run2018A-17Sep2018-v2/201018_142029/0000/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/EGamma/crab_EGamma_Run2018A-17Sep2018-v2/201018_142452/0000/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/EGamma/crab_EGamma_Run2018A-17Sep2018-v2/201018_142452/0001/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/EGamma/crab_EGamma_Run2018A-17Sep2018-v2/201018_142452/0002/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/EGamma/crab_EGamma_Run2018A-17Sep2018-v2/201018_142452/0003/*.root");
  chain->Add("/pnfs/ihep.ac.cn/data/cms/store/user/guoj/2018data/UFHZZAnalysisRun2/myTask_Data/MuonEG/crab_MuonEG_Run2018A-17Sep2018-v1/201018_142240/0000/*.root");

  long64_t nentries = chain->GetEntries();
  lep_id = 0;
  lep_pt = 0;
  lep_eta = 0;
  lep_phi = 0;
  lep_mass = 0;
  lepFSR_pt = 0;
  lepFSR_eta = 0;
  lepFSR_phi = 0;
  lepFSR_mass = 0;
  Z_pt = 0;
  Z_eta = 0;
  Z_phi = 0;
  Z_mass = 0;
  Z_noFSR_pt = 0;
  Z_noFSR_eta = 0;
  Z_noFSR_phi = 0;
  Z_noFSR_mass = 0;
  H_pt = 0;
  H_eta = 0;
  H_phi = 0;
  H_mass = 0;

  chain->SetBranchAddress("passedZ1LSelection",&passedZ1LSelection);
  chain->SetBranchAddress("passedZ4lSelection",&passedZ4lSelection);
  chain->SetBranchAddress("passedFullSelection",&passedFullSelection);
  chain->SetBranchAddress("passedTrig", &passedTrig);
  chain->SetBranchAddress("finalState",&finalState);
  chain->SetBranchAddress("lep_Hindex",&lep_Hindex);
  chain->SetBranchAddress("lep_id", &lep_id);
  chain->SetBranchAddress("lep_pt", &lep_pt);
  chain->SetBranchAddress("lep_eta", &lep_eta);
  chain->SetBranchAddress("lep_phi", &lep_phi);
  chain->SetBranchAddress("lep_mass", &lep_mass);
  chain->SetBranchAddress("lepFSR_pt", &lepFSR_pt);
  chain->SetBranchAddress("lepFSR_eta", &lepFSR_eta);
  chain->SetBranchAddress("lepFSR_phi", &lepFSR_phi);
  chain->SetBranchAddress("lepFSR_mass", &lepFSR_mass);
  chain->SetBranchAddress("Z_pt", &Z_pt);
  chain->SetBranchAddress("Z_eta", &Z_eta);
  chain->SetBranchAddress("Z_phi", &Z_phi);
  chain->SetBranchAddress("Z_mass", &Z_mass);
  chain->SetBranchAddress("Z_noFSR_pt", &Z_noFSR_pt);
  chain->SetBranchAddress("Z_noFSR_eta", &Z_noFSR_eta);
  chain->SetBranchAddress("Z_noFSR_phi", &Z_noFSR_phi);
  chain->SetBranchAddress("Z_noFSR_mass", &Z_noFSR_mass);
  chain->SetBranchAddress("H_pt",&H_pt);
  chain->SetBranchAddress("H_eta",&H_eta);
  chain->SetBranchAddress("H_phi",&H_phi);
  chain->SetBranchAddress("H_mass",&H_mass);

  for(long64_t i == 0; i<nentries; i++){
    chain->GetEntry(i);
    if (!passedTrig)    continue;
    newtree->Fill();
  }

  ofile.cd();
  newtree->Write();
  ofile.Close();

}

File: Getvariables.py
#!/usr/bin/env python
import ROOT
import os
import numpy as np
from array import array

import argparse
parser = argparse.ArgumentParser(description = "A simple ttree plotter")
parser.add_argument("-t", "--ttree", dest="ttree", default="Ana/passedEvents", help="TTree Name")
parser.add_argument("-i", "--inputfiles", dest="inputfiles", default="Sync_1031_2018_ttH_v2.root", help="List of input root files")
parser.add_argument("-o", "--outputfile", dest="outputfile", default="plots.root", help="Output file containing plots")
parser.add_argument("-s","--substring", dest="substring",default="",help='only submit datasets with this string in the name')
args = parser.parse_args()

print("get file " + str(args.inputfiles))
#input Ntuple
chain = ROOT.TChain(args.ttree)
f = open(args.inputfiles)
for line in f:
    dataset = line.strip('\n')
    print("dataset after strip " + str(dataset))
    chain.Add(dataset)
print 'Total number of events: ' + str(chain.GetEntries())

#variables
lep1_pt = array('f',[0.])
'''
lep_Hindex = array('l'[0])
lep_id = array('l',[0])
lep_pt = array('f',[0.])
lep_phi = array('f',[0.])
lep_eta = array('f',[0.])
lep_mass = array('f',[0.])
lepFSR_pt = array('f',[0.])
lepFSR_phi = array('f',[0.])
lepFSR_eta = array('f',[0.])
lepFSR_mass = array('f',[0.])
Z_pt = array('f',[0.])
Z_eta = array('f',[0.0])
Z_phi = array('f',[0.])
Z_mass = array('f',[0.])
Z_noFSR_pt = array('f',[0.])
Z_noFSR_eta = array('f',[0.0])
Z_noFSR_phi = array('f',[0.])
Z_noFSR_mass = array('f',[0.])
H_pt = array('f',[0.])
H_eta = array('f',[0.])
H_phi = array('f',[0.])
H_mass = array('f',[0.])
passedFullSelection = True
passedTrig = True
passedZ1LSelection = True
passedZ4lSelection = True
'''

#Output file and any Branch we want
file_out = ROOT.TFile(args.outputfile, 'recreate')
passedEvents = ROOT.TTree("passedEvents","passedEvents")
passedEvents.Branch("lep_pt1",lep1_pt,"lep1_pt/F")
'''
passedEvents.Branch("lep_Hindex",lep_Hindex,"lep_Hindex/L")
passedEvents.Branch("lep_id",lep_id,"lep_id/L")
passedEvents.Branch("lep_pt",lep_pt,"lep_pt/F")
passedEvents.Branch("lep_eta",lep_eta,"lep_eta/F")
passedEvents.Branch("lep_phi",lep_phi,"lep_phi/F")
passedEvents.Branch("lep_mass",lep_mass,"lep_mass/F")
passedEvents.Branch("lepFSR_pt",lepFSR_pt,"lepFSR_pt/F")
passedEvents.Branch("lepFSR_eta",lepFSR_eta,"lepFSR_eta/F")
passedEvents.Branch("lepFSR_phi",lepFSR_phi,"lepFSR_phi/F")
passedEvents.Branch("lepFSR_mass",lepFSR_mass,"lepFSR_mass/F")
passedEvents.Branch("Z_pt",Z_pt,"Z_pt/F")
passedEvents.Branch("Z_eta",Z_eta,"Z_eta/F")
passedEvents.Branch("Z_phi",Z_phi,"Z_phi/F")
passedEvents.Branch("Z_mass",Z_mass,"Z_mass/F")
passedEvents.Branch("Z_noFSR_pt",Z_noFSR_pt,"Z_noFSR_pt/F")
passedEvents.Branch("Z_noFSR_eta",Z_noFSR_eta,"Z_noFSR_eta/F")
passedEvents.Branch("Z_noFSR_phi",Z_noFSR_phi,"Z_noFSR_phi")
passedEvents.Branch("Z_noFSR_mass",Z_noFSR_mass,"Z_noFSR_mass")
passedEvents.Branch("H_pt",H_pt,"H_pt/F")
passedEvents.Branch("H_eta",H_eta,"H_eta/F")
passedEvents.Branch("H_phi",H_phi,"H_phi/F")
passedEvents.Branch("H_mass",H_mass,"H_mass")
passedEvents.Branch("passedTrig",passedTrig,"passedTrig/O")
passedEvents.Branch("passedZ1LSelection",passedZ1LSelection,"passedZ1LSelection/O")
passedEvents.Branch("passedZ4lSelection",passedZ4lSelection,"passedZ4lSelection/O")
passedEvents.Branch("passedFullSelection",passedFullSelection,"passedFullSelection/O")
'''

#Loop over all the events in the input ntuple
for ievent,event in enumerate(chain):
    Nlep = event.lep_pt.size()
    for i in range(Nlep):
    #fill tree
       lep1_pt[0] = event.lep_pt[event.lep_Hindex[0]]
#      lep_Hindex[i] = event.lep_Hindex[i]
       passedEvents.Fill()

file_out.Write()
file_out.Close()
